packages/core/plugin/src/withSentryIOS.ts

```diff
@@ -79,3 +85,58 @@ export function addSentryWithBundledScriptsToBundleShellScript(script: string):
     (match: string) => `/bin/sh ${SENTRY_REACT_NATIVE_XCODE_PATH} ${match}`,
   );
 }
+
+export function modifyAppDelegate(config: ExpoConfig): ExpoConfig {
+  return withAppDelegate(config, async config => {
+    if (!config.modResults || !config.modResults.path) {
+      warnOnce('Skipping AppDelegate modification because the file does not exist.');
+      return config;
+    }
+
+    const fileName = path.basename(config.modResults.path);
+
+    if (config.modResults.language === 'swift') {
+      if (config.modResults.contents.includes('RNSentrySDK.start()')) {
+        warnOnce(`Your '${fileName}' already contains 'RNSentrySDK.start()'.`);
+        return config;
+      }
+      if (!config.modResults.contents.includes('import RNSentry')) {
+        // Insert import statement after UIKit import
+        config.modResults.contents = config.modResults.contents.replace(/(import UIKit\n)/, `$1import RNSentry\n`);
+      }
+      // Add RNSentrySDK.start() at the beginning of application method
+      const originalContents = config.modResults.contents;
+      config.modResults.contents = config.modResults.contents.replace(
+        /(func application\([^)]*\) -> Bool \{)/s,
+        `$1\n    RNSentrySDK.start()`,
+      );
+      if (config.modResults.contents === originalContents) {
+        warnOnce(`Failed to insert 'RNSentrySDK.start()'.`);
+      }
+    } else {
+      // Objective-C
```

Comment:
Can we explicitly check for obj-c?

Reply:
Updated with a53c7f46a14361e8f3b2b1da7f52b4f404799751